aggregator/src/aggregator/test_util.rs

```diff
@@ -154,3 +161,37 @@ pub fn generate_helper_report_share_for_plaintext(
         .unwrap(),
     )
 }
+
+pub async fn assert_task_aggregation_counter(
+    datastore: &Datastore<MockClock>,
+    task_id: TaskId,
+    expected_counters: TaskAggregationCounter,
+) {
+    // We can't coordinate with the counter-update tasks, so we loop on polling them.
+
+    let end_instant = Instant::now() + Duration::from_secs(10);
+    loop {
+        let counters = datastore
+            .run_unnamed_tx(|tx| {
+                Box::pin(async move {
+                    Ok(tx
+                        .get_task_aggregation_counter(&task_id)
+                        .await
+                        .unwrap()
+                        .unwrap())
+                })
+            })
+            .await
+            .unwrap();
+
+        if counters == expected_counters {
+            return;
+        }
+        if Instant::now() > end_instant {
```

Comment:
I think we could make this a bit more resilient if we snapshotted `Instant::now()` at the top of the loop. That way, if fetching the counters the first time takes way too long due to resource exhaustion from other tests, or locks taken by database maintenance processes, then we will get a second chance to read from the database.

Reply:
I suppose -- in my testing, this loop often terminated on the first attempt, and failing that, nearly always the second. If 10s is not enough time in a unit test, something has gone pretty wrong.